presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/FairSharingQueue.java

```diff
@@ -0,0 +1,196 @@
+/*
```

Comment:
Please add unit tests for this new queue implementation. There is a bunch of tests for the other queue impls.